statefulset/info.go

```diff
@@ -0,0 +1,72 @@
+package statefulset
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/vshn/statefulset-resize-controller/pvc"
+	appsv1 "k8s.io/api/apps/v1"
+)
+
+// FailedLabel is a label for failed sts resizing that need human interaction
+const FailedLabel = "sts-resize.appuio.ch/failed"
+
+// PvcAnnotation is an annotation in which the initial state of the pvcs is stored in
+const PvcAnnotation = "sts-resize.appuio.ch/pvcs"
+
+// Info contains all data to manage a statfulset resizing
+type Info struct {
+	Old  *appsv1.StatefulSet
+	Pvcs []pvc.Info
+
+	sts *appsv1.StatefulSet
+}
```

Comment:
I can see that `Info` is a little generic, but I don't see how `tuple` is any better? And why is it a problem that there is `pvc.Info`, wouldn't that make `statefulset.Info` a more consistent naming then `statefulset.Tuple`?

Reply:
Tuple is often used in programming as a form to contain multiple closely related objects into one parameter or return type. Another name would be `record` or `row`, but those may not be the best ones here. When programmers read tuple, there's often the expectation of just that: holding together.
I don't know if Tuple is a good name for the PVC case. But I don't like `Info` either...
Probably `Context` fits better.

Reply:
`Tuple` for me makes the impression that it only holds data. But in this case it also has methods defined on it, which would be counter intuitive. It doesn't really just hold two related objects. It holds the StatefulSet and its PVCs and defines methods on the statfulset.

I'm not sure if I like `Context` either, as it has a very fixed definition in Go. It would also mean that we would often pass two different contexts, which I think is just confusing.

Maybe `pvc.Object` and `statefulset.Object`? Or `pvc.Entity` and `statefulset.Entity`

Reply:
`Entity` is LGTM. It conveys that it's business relevant, in which they are.